artifactory/cli.go

```diff
@@ -621,15 +621,98 @@ func GetCommands() []cli.Command {
 	}
 }
 
-func getGlobalConfigFlag() []cli.Flag {
+func getBaseBuildToolsConfigFlags() []cli.Flag {
 	return []cli.Flag{
 		cli.BoolFlag{
 			Name:  "global",
-			Usage: "[Default: false] Set to true, if you'd like to configuration to be global (for all projects). Specific projects can override the global configuration.` `",
+			Usage: "[Default: false] Set to true if you'd like the configuration to be global (for all projects). Specific projects can override the global configuration.` `",
+		},
+		cli.BoolTFlag{
+			Name:  "interactive",
+			Usage: "[Default: true] Set to false if you do not want the config command to be interactive.` `",
+		},
+		cli.BoolTFlag{
+			Name:  "resolveFromArtifactory",
+			Usage: "[Default: true] Set to false if you do not wish to resolve dependencies from Artifactory.` `",
+		},
+		cli.StringFlag{
+			Name:  "resolutionServerId",
+			Usage: "[Optional] Artifactory server ID for resolution.` `",
+		},
+		cli.BoolTFlag{
+			Name:  "deployToArtifactory",
+			Usage: "[Default: true] Set to false if you do not wish to deploy artifacts to Artifactory.` `",
+		},
+		cli.StringFlag{
+			Name:  "deploymentServerId",
+			Usage: "[Optional] Artifactory server ID for deployment.` `",
 		},
 	}
 }
 
+func getCommonBuildToolsConfigFlags() []cli.Flag {
+	return append(getBaseBuildToolsConfigFlags(),
+		cli.StringFlag{
+			Name:  "resolutionRepo",
+			Usage: "[Optional] Resolution repository.` `",
+		},
+		cli.StringFlag{
+			Name:  "deploymentRepo",
+			Usage: "[Optional] Deployment repository.` `",
+		},
+	)
+}
+
+func getMavenConfigFlags() []cli.Flag {
+	return append(getBaseBuildToolsConfigFlags(),
+		cli.StringFlag{
+			Name:  "resolutionReleaseRepo",
+			Usage: "[Optional] Resolution release repository.` `",
+		},
+		cli.StringFlag{
+			Name:  "resolutionSnapshotRepo",
+			Usage: "[Optional] Resolution snapshot repository.` `",
+		},
+		cli.StringFlag{
+			Name:  "deploymentReleaseRepo",
+			Usage: "[Optional] Deployment release repository.` `",
+		},
+		cli.StringFlag{
+			Name:  "deploymentSnapshotRepo",
+			Usage: "[Optional] Deployment snapshot repository.` `",
```

Comment:
Deployment snapshot repository. --> Deployment repository for snapshot artifacts..